knative-operator/pkg/controller/knativeserving/kourier/kourier.go

```diff
@@ -33,28 +33,19 @@ func Apply(instance *servingv1alpha1.KnativeServing, api client.Client, scheme *
 	if err != nil {
 		return err
 	}
-	if instance.Status.IsFullySupported() {
-		// TODO: verify deployed kourier is not different from kourier-latest.yaml accurately.
-		if err := checkDeployments(&manifest, instance, api); err == nil {
-			return nil
-		}
-	}
-
-	// Us reaching here means we need to do something and/or wait longer.
-	instance.Status.MarkDependencyInstalling("Kourier")
-	if err := api.Status().Update(context.TODO(), instance); err != nil {
-		return err
-	}
-
 	log.Info("Installing Kourier Ingress")
 	if err := manifest.Apply(); err != nil {
 		return err
 	}
 	if err := checkDeployments(&manifest, instance, api); err != nil {
+		instance.Status.MarkDependencyInstalling("Kourier")
```

Comment:
Do we need to guard this further by checking if we're already in this state? I think calling this will cause us to update the "LastTransitionTimestamp" all the time, which in turn will cause update conflicts.

Reply:
Sure, I think it should be guarded. Updated.

Reply:
I think this was nonsense from my end.

The SetConditions function underneath looks like this

```golang
// SetCondition sets or updates the Condition on Conditions for Condition.Type.
// If there is an update, Conditions are stored back sorted.
func (r conditionsImpl) SetCondition(new Condition) {
	if r.accessor == nil {
		return
	}
	t := new.Type
	var conditions Conditions
	for _, c := range r.accessor.GetConditions() {
		if c.Type != t {
			conditions = append(conditions, c)
		} else {
			// If we'd only update the LastTransitionTime, then return.
			new.LastTransitionTime = c.LastTransitionTime
			if reflect.DeepEqual(&new, &c) {
				return
			}
		}
	}
	new.LastTransitionTime = VolatileTime{Inner: metav1.NewTime(time.Now())}
	conditions = append(conditions, new)
	// Sorted for convenience of the consumer, i.e. kubectl.
	sort.Slice(conditions, func(i, j int) bool { return conditions[i].Type < conditions[j].Type })
	r.accessor.SetConditions(conditions)
}
```

Note the early exit on "If we'd only update the LastTransitionTime then return". So we're actually good without a guard. Sorry for the noise.